frontend/src/pages/docs/[...slug].tsx

```diff
@@ -1,75 +1,162 @@
+import { GetStaticPaths } from "next";
+import { MDXRemote, MDXRemoteSerializeResult } from "next-mdx-remote";
+import { serialize } from "next-mdx-remote/serialize";
+import Image from "next/image";
+import NextLink from "next/link";
+import pathTool from "path";
+import { memo } from "react";
+import styled from "styled-components";
+
+const DOC_SITE_FOLDER_NAME = "doc-site";
+
+interface Directory {
+  dirName: string;
+  files: Array<string>;
+  slug: Array<string>;
+  subDirectories: Array<Directory>;
+}
+
+const CACHED_FILE_PATHS = new Map<string, Directory>();
+function filePaths(...root: Array<string>): Directory {
+  const cacheStore = CACHED_FILE_PATHS && CACHED_FILE_PATHS.get(root.join("/"));
+  if (cacheStore) {
+    return cacheStore;
+  }
+
+  const newDirectory = {
+    dirName: root[root.length - 1] ?? "",
+    files: [],
+    slug: root,
+    subDirectories: [],
+  } as Directory;
+
+  const paths = fs.readdirSync(pathTool.join(DOC_SITE_FOLDER_NAME, ...root));
+  paths.forEach((path) => {
+    const builtFullPath = pathTool.join(DOC_SITE_FOLDER_NAME, ...root, path);
+    if (fs.lstatSync(builtFullPath).isFile()) {
+      if (path.endsWith(".mdx"))
+        newDirectory.files.push(path.replace(".mdx", ""));
+    } else {
+      newDirectory.subDirectories.push(filePaths(...root, path));
+    }
+  });
+  CACHED_FILE_PATHS.set(root.join("/"), newDirectory);
+  return newDirectory;
+}
+
+function generatePaths(
+  directory: Directory,
+  slugs = new Array<{ params: { slug: Array<string> } }>()
+): Array<{ params: { slug: Array<string> } }> {
+  directory.files.forEach((file) => {
+    slugs.push({ params: { slug: [...directory.slug, file] } });
+  });
+  if (directory.subDirectories.length > 0) {
+    directory.subDirectories.forEach((subDirectory) => {
+      slugs.push(...generatePaths(subDirectory, slugs));
+    });
+  }
+  return slugs;
+}
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const filePath = filePaths();
+  const paths = generatePaths(filePath);
+
+  return {
+    fallback: false,
+    paths,
+  };
+};
+
+export const getStaticProps = async ({
+  params: { slug },
+}: {
+  params: { slug: Array<string> };
+}): Promise<{
+  props: Props;
+}> => {
+  const markdownWithMeta = fs.readFileSync(
+    pathTool.join("doc-site", slug.join("/") + ".mdx"),
+    "utf-8"
+  );
+
+  const filePath = filePaths();
+  const { data: frontMatter, content } = matter(markdownWithMeta);
+  const mdxSource = await serialize(content);
+  return {
+    props: {
+      filePath,
+      frontMatter,
+      mdxSource,
+      slug,
+    },
+  };
+};
+
+const Directory = memo(function RenderDirecotry({
+  directory,
+}: {
+  directory: Directory;
+}) {
+  return (
+    <ul>
+      {directory.files.map((file) => {
+        let href = "/docs/";
+        if (directory.slug.length > 0) href += directory.slug.join("/") + "/";
+        href += file;
+        return (
+          <NextLink key={file} href={href}>
```

Comment:
I think we need `passHref` prop to pass the URL, otherwise the `<a />` link doesn't get `href`!

<img width="1181" alt="Screen Shot 2022-05-23 at 2 42 54 PM" src="https://user-images.githubusercontent.com/6309723/169910515-d38effb8-f02e-48bb-96d3-145b6bab8fd6.png">